tools/bridge/end2end-tests/tests/test_transfer.py

```diff
@@ -485,3 +522,186 @@ def check_balance():
         assert home_balance_before - home_balance_after == -value
 
     assert_within_timeout(check_balance, 10)
+
+
+def test_offline_validators_validates_not_complete_transfer(
+    web3_home, web3_foreign, foreign_bridge_contract, token_contract, accounts, bridges
+):
+    """
+    Tests whether a transfer initiated when threshold validators are offline will be effective when enough validators turn online
+    """
+    bridges[0].start()
+
+    sender = accounts[3]
+    value = 5000
+    foreign_balance_before = token_contract.functions.balanceOf(sender).call()
+    home_balance_before = web3_home.eth.getBalance(sender)
+
+    wait_for_successful_transaction_receipt(
+        web3_foreign,
+        token_contract.functions.transfer(
+            foreign_bridge_contract.address, value
+        ).transact({"from": sender}),
+    )
+
+    mine_min_blocks(web3_foreign, REORG_DEPTH)
+
+    def check_balances_bridge_transfer_not_complete():
+        foreign_balance_transfer_not_complete = token_contract.functions.balanceOf(
+            sender
+        ).call()
+        home_balance_transfer_not_complete = web3_home.eth.getBalance(sender)
+
+        assert foreign_balance_before - foreign_balance_transfer_not_complete == value
+        assert home_balance_before == home_balance_transfer_not_complete
+
+    assert_within_timeout(check_balances_bridge_transfer_not_complete, 10)
```

Comment:
This should be assert after timeout, otherwise it could be that you checked just to early.